### PR TITLE
HAI-2189 Create tables for application customers and contacts

### PR DESCRIPTION
**Pull request by corvidian**
# Description

Create tables for application customers and contacts. Also, create entities for them.

## Type of change

- [ ] Bug fix 
- [X] New feature 
- [ ] Other

### CODE REVIEW

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    var nimi: String,
+    var sahkoposti: String?,
+    var puhelinnumero: String?,
+    var ytunnus: String?,
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "application_id")
+    var application: ApplicationEntity,
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "hakemusyhteystietoEntity")
+    @BatchSize(size = 100)
+    var yhteystiedot: List<HakemusyhteyshenkiloEntity>,
```
Comment (umeetiusbaar) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt:34]: Should this be `yhteyshenkilot` instead?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@Entity
+@Table(name = "hakemusyhteyshenkilo")
+class HakemusyhteyshenkiloEntity(
+    @Id val id: UUID = UUID.randomUUID(),
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "hakemusyhteystieto_id")
+    var hakemusyhteystietoEntity: HakemusyhteystietoEntity,
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "hankekayttaja_id")
+    var hankekayttajaEntity: HankekayttajaEntity,
```
Comment (umeetiusbaar) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt:46]: Should we follow the naming in `HankeYhteyshenkiloEntity`? E.g. `hakemusyhteystieto` and `hankekayttaja`? (I do like the correct spelling of these compound words).

File: services/hanke-service/src/main/resources/db/changelog/changesets/070-create-hakemusyhteystieto-tables.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+COMMENT ON COLUMN hakemusyhteystieto.puhelinnumero IS 'Telephone number for the customer.';
+COMMENT ON COLUMN hakemusyhteystieto.y_tunnus IS 'Business ID of the customer if the customer is an organization.';
+
+create table hakemusyhteyshenkilo
+(
+    id                    uuid    not null primary key,
+    hakemusyhteystieto_id uuid    not null,
+    hankekayttaja_id      uuid    not null,
+    tilaaja               boolean not null,
+    constraint fk_hakemusyhteyshenkilo_hakemusyhteystieto foreign key (hakemusyhteystieto_id) references hakemusyhteystieto (id)
```
Comment (umeetiusbaar) [inline services/hanke-service/src/main/resources/db/changelog/changesets/070-create-hakemusyhteystieto-tables.sql:35]: Should there also be FK pointing to hankekayttaja as there is in `hankeyhteyshenkilo` table?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt (hunk 2 above)
Comment (umeetiusbaar) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt:46]: Should `HankekayttajaEntity` have `hakemusyhteyshenkilot` list as it now have `(hanke)yhteyshenkilot` list?

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt (hunk 1 above)
Comment (corvidian) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt]: Yes, it should. Thank you for noticing. Renamed.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt (hunk 2 above)
Comment (corvidian) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt]: > Should we follow the naming in `HankeYhteyshenkiloEntity`? E.g. `hakemusyhteystieto` and `hankekayttaja`? (I do like the correct spelling of these compound words).

Yes, we should. Thank you for noticing. Renamed.

Comment (corvidian) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt]: > Should `HankekayttajaEntity` have `hakemusyhteyshenkilot` list as it now have `(hanke)yhteyshenkilot` list?

It shouldn't hurt. Added.

File: services/hanke-service/src/main/resources/db/changelog/changesets/070-create-hakemusyhteystieto-tables.sql (hunk 3 above)
Comment (corvidian) [reply services/hanke-service/src/main/resources/db/changelog/changesets/070-create-hakemusyhteystieto-tables.sql]: Yes, there definitely should. Thank you for noticing. I added the foreign key.

**Review by umeetiusbaar — APPROVED** [review]

**Comment by umeetiusbaar** [comment]
Sonarcloud complains about missing varchar sizes, is that a problem?

**Pull request merged**